pkg/service/scheduler/model.go

```diff
@@ -70,6 +70,31 @@ func (t *TaskType) UnmarshalText(text []byte) error {
 	return nil
 }
 
+// AllowedTaskType is a wrapper around TaskType that allows it to be empty and adds additional validation.
+type AllowedTaskType struct {
+	TaskType
+}
+
+// IsEmpty returns true if the TaskTypeOptional is empty.
+func (t AllowedTaskType) IsEmpty() bool {
+	return string(t.TaskType) == ""
+}
+
+func (t *AllowedTaskType) UnmarshalText(text []byte) error {
+	if len(text) == 0 {
+		return nil
+	}
+	tt := TaskType(text)
+	if tt == SuspendTask || tt == HealthCheckTask {
+		return fmt.Errorf("AllowedTaskType cannot be %q", tt)
```

Comment:
nit: this is a user facing error, so maybe `"AllowedTaskType cannot be %q` -> `"allowed task type cannot be %q` or `"allowed_task_type cannot be %q`?